cpp/collections_module/algorithm/algorithm.cpp

```diff
@@ -0,0 +1,37 @@
+#include "algorithm.hpp"
+
+#include <unordered_set>
+#include "mgp.hpp"
+
+void Collections::Intersection(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto &list1{arguments[0].ValueList()};
+    std::unordered_set<mgp::Value> set1{list1.begin(), list1.end()};
+    const auto &list2{arguments[1].ValueList()};
+    std::unordered_set<mgp::Value> set2{list2.begin(), list2.end()};
+
+    if (set1.size() > set2.size()) {
+      auto tmp(std::move(set1));
+      set1 = std::move(set2);
+      set2 = std::move(tmp);
+    }
+
+    mgp::List intersection{};
+    for (const auto &element : set1) {
+      if (set2.find(element) != set2.end()) {
```

Comment:
You should be able to use `contains` here in my opinion, as you are not doing anything with `find`